subsys/bluetooth/mesh/Kconfig

```diff
@@ -289,6 +289,46 @@ config BT_MESH_CRPL
 	  protection list. This option is similar to the network message
 	  cache size, but has a different purpose.
 
+choice BT_MESH_RPL_STORAGE_MODE
+	prompt "Replay protection list storage mode"
+	default BT_MESH_RPL_STORAGE_MODE_SETTINGS
+
+config BT_MESH_RPL_STORAGE_MODE_SETTINGS
+	bool "Persistent storage of RPL in Settings"
+	help
```

Comment:
It might seem a little akward to not add this dependency, but the idea is that this is **not** dependent on BT_SETTINGS. Like described in the help text for this option, if `BT_SETTINGS != y` rpl.c will provide a non-persistent alternative for RPL (Just like the previous implementation).

Reply:
It could be solved by something like this:
```
choice BT_MESH_RPL_STORAGE_MODE
	prompt "Replay protection list storage mode"
	default BT_MESH_RPL_STORAGE_MODE_SETTINGS

config BT_MESH_RPL_STORAGE_MODE_SETTINGS
	bool "Persistent storage of RPL in Settings"
	depends on BT_SETTINGS
	help
	  Persistent storage of RPL in Settings. 

config BT_MESH_RPL_STORAGE_MODE_NON_PERSIST
	bool "Non-persistent storage of RPL "
	depends on !BT_SETTINGS
	help
	  Storage of RPL in volatile memory only.

endchoice
```

```
if(CONFIG_BT_MESH_RPL_STORAGE_MODE_SETTINGS OR CONFIG_BT_MESH_RPL_STORAGE_MODE_NON_PERSIST)
  zephyr_library_sources(rpl.c)
endif()
```
But I think that the added bloat and complexity might not justify it.